src/wallet/coin_selection.rs

```diff
@@ -1148,23 +1129,23 @@ mod test {
     fn test_bnb_coin_selection_check_fee_rate() {
         let utxos = get_test_utxos();
         let database = MemoryDatabase::default();
+        let target_amount = 99932; // first utxo's effective value
 
         let result = BranchAndBoundCoinSelection::new(0)
             .coin_select(
                 &database,
                 vec![],
                 utxos,
                 FeeRate::from_sat_per_vb(1.0),
-                99932, // first utxo's effective value
-                0,
+                target_amount,
             )
             .unwrap();
 
         assert_eq!(result.selected.len(), 1);
         assert_eq!(result.selected_amount(), 100_000);
         let input_size = (TXIN_BASE_WEIGHT + P2WPKH_WITNESS_SIZE).vbytes();
-        let epsilon = 0.5;
-        assert!((1.0 - (result.fee_amount as f32 / input_size as f32)).abs() < epsilon);
+        // the final fee rate should be exactly the same that the fee rate given
```

Comment:
I thought that as we were using the first utxo's effective value as target amount an from this PR on we don't have to consider the accumulated fee amount of the overall transaction, it was a good way to illustrate the idea that BnB would produce an exact match. I don't have strong opinions on this, so I can roll it back.

Reply:
Maybe it's just me but I always try to avoid comparing floats with strict equality because there's inherent rounding happening and I'm always afraid it will cause something to break sooner or later.

Maybe in this case it won't happen, but just to be safe I would keep an epsilon, even if very small.

Reply:
It does seem that this test should be for "an exact match" whilst taking FP rounding errors into consideration. I also agree that `0.5` is probably too big ([machine epsilon for `binary32` is like 24 decimal places](https://en.wikipedia.org/wiki/Machine_epsilon) which roughly means, based on my limited understanding, numbers being stored with 23-24 digits of precision).

I would say, having `epsilon` as `0.01` should be plenty lol (no point going any smaller imo).

Reply:
I found there is a constant defined for `f32` machine epsilon, probably for this kind of cases. I'll use it instead of `0.01`.